### PR TITLE
Remove the orderField

### LINKED ISSUE #4380: Remove or deprecate orderField for fileTree as well
### Description

As discussed when reviewing #4367 we should remove/deprecate the `orderField` from the gallery file tree too. This can be done by including both, directory and file hashes in the multiSRC field (in the correct order).

### PR DESCRIPTION
**Pull request by ausi**
Fixes #4380 

I’d like to remove the `orderField` completely in Contao 5.0 and backport the support for omitting the orderField to Contao 4.13: #4752 
@contao/developers WDYT?

### CODE REVIEW

File: core-bundle/src/Migration/Version500/OrderFieldMigration.php [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+use Contao\CoreBundle\Migration\AbstractMigration;
+use Contao\CoreBundle\Migration\MigrationResult;
+use Contao\StringUtil;
+use Doctrine\DBAL\Connection;
+
+/**
+ * @internal
+ */
+class OrderFieldMigration extends AbstractMigration
```
Comment (m-vo) [inline core-bundle/src/Migration/Version500/OrderFieldMigration.php:23]: Should this be another abstract migration that people could reuse for their tables/fields?

Comment (ausi) [reply core-bundle/src/Migration/Version500/OrderFieldMigration.php:23]: Don’t think so, the first `Version410\OrderFieldMigration` is also not abstract.

**Review by leofeyer — DISMISSED** [review]
Perfect 👍

File: core-bundle/src/Migration/Version500/OrderFieldMigration.php (hunk 1 above)
Comment (Toflar) [reply core-bundle/src/Migration/Version500/OrderFieldMigration.php:23]: Imho both would make sense, no? :)

**Review by leofeyer — APPROVED** [review]

File: core-bundle/src/Migration/Version500/OrderFieldMigration.php (hunk 1 above)
Comment (leofeyer) [reply core-bundle/src/Migration/Version500/OrderFieldMigration.php:23]: We should probably create a different PR that adds the abstract migration and adjusts the two existing ones.

**Comment by leofeyer** [comment]
Thank you @ausi.

**Pull request merged**